### PR TITLE
Bump up CUDA version used for tests

### PR DESCRIPTION
**Pull request by JanuszL**
- disable MXNet tests are MXNet is no longer
  actively developed
- extend support for DLPack capsules that provide
  strides collapsing dimensions
- updates TSM PaddlePaddle example to the latest API
- removes SSD PaddlePaddle example as it uses many no longer
  supported functions (SSD specific)

## Category:
**Other** (*e.g. Documentation, Tests, Configuration*)

## Description:
- disable MXNet tests are MXNet is no longer
  actively developed
- extend support for DLPack capsules that provide
  strides collapsing dimensions
- updates TSM PaddlePaddle example to the latest API
- removes SSD PaddlePaddle example as it uses many no longer
  supported functions (SSD specific)

### Affected modules and functionalities:
- qa/setup_packages.py

### Tests:

- [x] Existing tests apply
  - all python related
- [ ] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [ ] N/A

### Documentation
- [ ] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [x] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [x] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

### CODE REVIEW

File: dali/python/backend_impl.cc [hunk 1]
```diff
@@ -155,11 +155,18 @@ void CheckContiguousTensor(const TStrides &strides, int num_strides,
...
   int64_t stride_from_shape = element_size;
+  int64_t stride_from_shape_collapsed = 1;
+  int64_t last_non_one_dim = 1;
   for (int i = num_strides - 1; i >= 0; i--) {
-    DALI_ENFORCE(strides[i] == stride_from_shape,
+    DALI_ENFORCE(strides[i] == stride_from_shape || strides[i] == stride_from_shape_collapsed,
         make_string("Strided data not supported. Dimension ", i, " has stride ", strides[i],
         " whereas densely packed data of this shape would have a stride ", stride_from_shape));
     stride_from_shape *= shape[i];
+    // for shapes [1, 1, 5] leading dimensions may not contribute to
```
Comment (stiepan) [inline dali/python/backend_impl.cc:165]:
```suggestion
    // for shapes [1, 1, 5] leading dimensions may not contribute to stride
```
?

File: dali/test/python/test_torch_pipeline_rnnt.py [hunk 2]
```diff
@@ -159,7 +159,9 @@ def forward(self, inp, seq_len):
         # do stft
         x = torch.stft(x, n_fft=self.n_fft, hop_length=self.hop_length,
                        win_length=self.win_length, pad_mode='reflect',
-                       center=True, window=self.window.to(dtype=torch.float).to(x.device))
+                       center=True, window=self.window.to(dtype=torch.float).to(x.device),
+                       return_complex=True)
+        x = torch.view_as_real(x)
```
Comment (stiepan) [inline dali/test/python/test_torch_pipeline_rnnt.py:164]: Why do we need this change?

File: docs/examples/use_cases/paddle/tsm/demo/7.mp4
Comment (stiepan) [inline docs/examples/use_cases/paddle/tsm/demo/7.mp4:1]: Do we need that file in the repo or is it artifact of running the example locally?

File: docs/examples/use_cases/paddle/tsm/utils.py [hunk 3]
```diff
@@ -92,9 +92,9 @@ def load_weights(exe, prog, url):
...
 
     if os.path.isdir(weight_path):
-        fluid.io.load_vars(
+        paddle.static.io.load_vars(
             exe, weight_path, prog,
             predicate=lambda v: os.path.exists(
                 os.path.join(weight_path, v.name)))
     else:
-        fluid.io.load_params(exe, '', prog, filename=weight_path)
+        paddle.distributed.io.load_persistables(exe, '', prog, filename=weight_path)
```
Comment (stiepan) [inline docs/examples/use_cases/paddle/tsm/utils.py:100]: nitpick: missing new line at the end of file

File: qa/TL0_FW_iterators/test_mxnet.sh [hunk 4]
```diff
@@ -1,6 +1,6 @@
 #!/bin/bash -e
 # used pip packages
-pip_packages='${python_test_runner_package} numpy mxnet'
+pip_packages='${python_test_runner_package} numpy'
```
Comment (stiepan) [inline qa/TL0_FW_iterators/test_mxnet.sh:3]:
```suggestion
# to be run inside a MXNet container - so don't need to list it here as a pip package dependency
pip_packages='${python_test_runner_package} numpy'
```

File: qa/TL1_custom_src_pattern_build/test.sh [hunk 5]
```diff
@@ -1,6 +1,7 @@
 #!/bin/bash -e
 
 build_and_check() {
+  pip install black dm-tree
```
Comment (stiepan) [inline qa/TL1_custom_src_pattern_build/test.sh:4]: Why do we need that?

File: qa/TL1_paddle_tsm/test.sh [hunk 6]
```diff
@@ -17,7 +17,7 @@ do_once() {
 
 test_body() {
   out=`python infer.py -k 1 -s 15 demo`
-  if echo $out | grep -E "(paragliding|sticking_tongue_out)"; then
+  if echo $out | grep -E "(waxing_legs|paragliding)"; then
```
Comment (stiepan) [inline qa/TL1_paddle_tsm/test.sh:20]: that's indeed a drastic change between the versions:)

File: dali/test/python/test_torch_pipeline_rnnt.py (hunk 2 above)
Comment (JanuszL) [reply dali/test/python/test_torch_pipeline_rnnt.py:164]: According to https://pytorch.org/docs/stable/generated/torch.stft.html:
```
From version 1.8.0, return_complex must always be given explicitly for real inputs and return_complex=False has been deprecated. Strongly prefer return_complex=True as in a future pytorch release, this function will only return complex tensors.

Note that [torch.view_as_real()](https://pytorch.org/docs/stable/generated/torch.view_as_real.html#torch.view_as_real) can be used to recover a real tensor with an extra last dimension for real and imaginary components.
```

File: qa/TL1_custom_src_pattern_build/test.sh (hunk 5 above)
Comment (JanuszL) [reply qa/TL1_custom_src_pattern_build/test.sh:4]: OK, we install this in the base test container.

Comment (stiepan) [reply qa/TL1_custom_src_pattern_build/test.sh:4]: I mean, I thought our `setup.py` already requires dm-tree. Isn't that relevent here?

Comment (JanuszL) [reply qa/TL1_custom_src_pattern_build/test.sh:4]: With stub generation you import DALI during the build without installing it, so your build env needs to have DALI dependencies beforehand. So `make -j` needs it already.

File: qa/TL0_FW_iterators/test_mxnet.sh (hunk 4 above)
Comment (JanuszL) [reply qa/TL0_FW_iterators/test_mxnet.sh:4]: Missed that. Fixed.

File: dali/python/backend_impl.cc (hunk 1 above)
Comment (JanuszL) [reply dali/python/backend_impl.cc]: Done

File: docs/examples/use_cases/paddle/tsm/demo/7.mp4
Comment (JanuszL) [reply docs/examples/use_cases/paddle/tsm/demo/7.mp4:1]: Leftover. Removed.

File: docs/examples/use_cases/paddle/tsm/utils.py (hunk 3 above)
Comment (JanuszL) [reply docs/examples/use_cases/paddle/tsm/utils.py]: Done

File: qa/TL1_custom_src_pattern_build/test.sh (hunk 5 above)
Comment (JanuszL) [reply qa/TL1_custom_src_pattern_build/test.sh]: Removed.

File: qa/TL1_paddle_tsm/test.sh (hunk 6 above)
Comment (JanuszL) [reply qa/TL1_paddle_tsm/test.sh:20]: In CI it returns `paragliding`, locally I get `waxing_legs`. Visually examining the clip doesn't give me much clue which one is more correct or both are equally not correct (and I get different result buy just updating PaddlePaddle and changing the model loading from `paddle.fluid.io.load_persistables(exe, '', prog, filename=weight_path)` to `paddle.distributed.io.load_persistables(exe, '', prog, filename=weight_path)`) :
image
image

File: qa/TL1_custom_src_pattern_build/test.sh (hunk 5 above)
Comment (stiepan) [reply qa/TL1_custom_src_pattern_build/test.sh]: I see, thanks!

**Review by stiepan — APPROVED** [review]

**Review by klecki — APPROVED** [review]

**Pull request merged**